cpp/include/cudf/aggregation.hpp

```diff
@@ -493,5 +495,74 @@ std::unique_ptr<Base> make_merge_sets_aggregation(null_equality nulls_equal = nu
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_merge_m2_aggregation();
 
+/**
+ * @brief Factory to create a TDIGEST aggregation
+ *
+ * Produces a tdigest (https://arxiv.org/pdf/1902.04023.pdf) column from input values.
+ * The input aggregation values are expected to be fixed-width numeric types.
+ *
+ * The tdigest column produced is of the following structure:
+ *
+ * struct {
+ *   // centroids for the digest
+ *   list {
+ *    struct {
+ *      double    // mean
+ *      double    // weight
+ *    }
+ *   }
+ *   // these are from the input stream, not the centroids. they are used
+ *   // during the percentile_approx computation near the beginning or
+ *   // end of the quantiles
+ *   double       // min
+ *   double       // max
+ * }
+ *
+ * @param delta Parameter controlling compression level and accuracy on subsequent
+ * queries on the output tdigest data.  Delta places an upper bound on the size of
+ * the computed tdigests: A delta of 1000 will result in a tdigest containing no
+ * more than 1000 centroids (32 bytes each). Higher values of delta result in more
+ * accurate tdigest information.
```

Comment:
`delta` feels like a non-intuitive name for this parameter. `max_centroids` seems more descriptive? 
```suggestion
 * @param max_centroids Upper bound on the number of centroids per tdigest. Higher values result in more accurate tdigest information. 
```

Reply:
Maybe.  The term `delta` is somewhat central to the overall tdigest definition from the external user's perspective, whereas "centroids" kind of isn't as they don't really know what exactly is going on under the hood.

@mythrocks  @revans2  @andygrove  For opinions.  I'm fine either way.

Reply:
I prefer `max_centroids`. delta is meant to

Reply:
Sounds good.  I'll change to `max_centroids` in the interface and leave it as delta in the guts of the generation code to keep it somewhat tied to the paper.